lxc/cookiejar/jar.go

```diff
@@ -176,3 +178,18 @@ func (j *Jar) Save() error {
 
 	return nil
 }
+
+// SetCookies implements [http.CookieJar.SetCookies]. It only accepts cookies from the remote host or its subdomains.
+func (j *Jar) SetCookies(u *url.URL, cookies []*http.Cookie) {
+	// Only accept cookies from the remote host or its subdomains.
+	// Note: This will still accept cookies from a subdomain on an IP address, e.g. sub.1.2.3.4
+	host := u.Hostname()
+	allowedHost := j.remote.Hostname()
+
+	if host == allowedHost || strings.HasSuffix(host, "."+allowedHost) {
+		// store cookies with the original URL (includes port)
+		j.Jar.SetCookies(u, cookies)
+	} else {
+		fmt.Printf("Rejected cookie from %s\n", host)
+	}
```

Comment:
Please just keep this to identical host names for now. If subdomains become a problem in the future, we can fix it then.

Also, please just ignore cookies from other hosts and don't print anything. We don't want to print to stdout as this might break shell scripts.